File: Algoritmica/Actividades/Actividad-I/Codigos-I/Tema1_3_Analisis/Tema1_26.py
def anagramaSolucion3(cadena1,cadena2):
    if len(cadena1)!=len(cadena2):
        return False

    unaLista1 = list(cadena1)
    unaLista2 = list(cadena2)

    unaLista1.sort()
    unaLista2.sort()

    pos = 0
    coincide = True

    while pos < len(cadena1) and coincide:
        if unaLista1[pos]==unaLista2[pos]:
            pos = pos + 1
        else:
            coincide = False

    return coincide

print(anagramaSolucion3('abcde','edcba'))

File: Algoritmica/Actividades/Actividad-I/Codigos-I/Tema1_3_Analisis/Tema1_26.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

// sorts anagrams in order from a-z, and then compares them
bool anagramsolution3(string s1, string s2)
{
    if (s1.length() != s2.length())
        return false;
    
    sort(s1.begin(), s1.end());
    sort(s2.begin(), s2.end());

    unsigned int pos = 0;
    bool matches = true;

    while (pos < s1.length() && matches)
        if (s1[pos] == s2[pos])
            pos = pos + 1;
        else
            matches = false;
    return matches;
}

int main()
{
    bool value = anagramsolution3("abcde", "edcba");
    if (value == 1)
        cout << "True" << endl;
    else
        cout << "False" << endl;
    return 0;
}
